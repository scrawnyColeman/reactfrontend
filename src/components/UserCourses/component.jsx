import React from 'react';
import styled from 'styled-components';
import Slider from 'react-slick';
import Button from '../Button/component';
import { colours, carouselSettings } from '../../constants/styles';
import { useHistory } from 'react-router-dom';
import { imageArray } from '../../images/imageLoader';

const Container = styled.div`
    margin: 0 auto;
    width: 45vw;
    height: 100%;
`;
const StyledCard = styled.div`
    display: inline-block;
    width: 95%;
    text-align: left;
    height: 490px;
    border: 2px ${colours.secondary};
`;
const StyledCardInformation = styled.div`
    height: 220px;
    width: 90%;
    margin: auto;
    display: grid;
    grid-template-rows: 50px 120px 35px;
    padding: 6px;
`;
const StyledCardHeader = styled.div`
    height: 270px
    overflow: hidden;
`;
const StyledImg = styled.img`
    width: 95%;
    height: 270px;
    margin: auto;
    object-fit: cover;
`;
const StyledLabel = styled.div`
    text-align: left;
    padding: 0 12px 12px;
    font-size: 1.75rem;
`;
const StyledDescription = styled.p`
    padding: 4px 0;
`;

const StyledHeading = styled.span`
    width: 100%;
    color: ${colours.primary};
    display: flex;
    justify-content: space-between;
    margin: 0 auto;
`;

const UserCourses = ({ courses }) => {
    const history = useHistory();

    return (
        <Container>
            <StyledLabel>Pick up where you left off...</StyledLabel>
            <Slider {...carouselSettings}>
                {courses.map((course) => {
                    const { lesson } = course;
                    const languageImage = imageArray.find((img) => img.path === lesson.language.imagePath);
                    return (
                        <StyledCard key={course.id}>
                            <StyledCardHeader>
                                <StyledImg src={languageImage.img} />
                            </StyledCardHeader>
                            <StyledCardInformation>
                                <StyledHeading>
                                    <p>{lesson.title}</p>
                                    <p>[{lesson.language.language.toUpperCase()}]</p>
                                </StyledHeading>
                                <StyledDescription>
                                    {lesson.description.slice(0, 150)}
                                    {lesson.description.length > 150 && '...'}
                                </StyledDescription>
                                <Button
                                    onClick={() => {
                                        history.push({ pathname: `/learn/${lesson.id}` });
                                    }}
                                    text="Go"
                                    size="small"
                                    variant="outlined"
                                    hierarchy="primary"
                                />
                            </StyledCardInformation>
                        </StyledCard>
                    );
                })}
            </Slider>
        </Container>
    );
};

export default UserCourses;
