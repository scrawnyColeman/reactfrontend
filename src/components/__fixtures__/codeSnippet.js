export const codeSnippet = {
    title: 'title',
    question: 'question',
    hint: 'hint here',
    codeSnippet: 'Code snippet.js',
    solution: 'answer',
};
