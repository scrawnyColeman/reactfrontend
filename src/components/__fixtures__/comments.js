export const commentArray = [
    {
        commentid: '1',
        comment: 'This is the best test',
        parentid: '0',
        id: '98',
        author: { userName: 'Testy McTestFace' },
    },
    {
        commentid: '2',
        comment: 'Im just happy to be here',
        parentid: '0',
        id: '99',
        author: { userName: 'Tester Zero' },
    },
];
export const childComments = [
    {
        commentid: '3',
        comment: 'This is the best test',
        parentid: '1',
        id: '100',
        author: { userName: 'Testy McTestFace' },
    },
    {
        commentid: '4',
        comment: 'Im just happy to be here',
        parentid: '1',
        id: '101',
        author: { userName: 'Tester Zero' },
    },
];
