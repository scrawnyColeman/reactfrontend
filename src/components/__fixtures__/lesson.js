export const lesson = {
    id: 1,
    title: 'Recursion',
    description:
        'Learn recursive functions to ace all of your coding interviews. We recommend that all users complete our loops and conditionals lesson before beginning here.',
    theory:
        'A common computer programming tactic is to divide a problem into sub-problems of the same type as the original, solve those sub-problems, and combine the results. This is often referred to as the divide-and-conquer method; when combined with a lookup table that stores the results of solving sub-problems (to avoid solving them repeatedly and incurring extra computation time), it can be referred to as dynamic programming or memoization.\\n\r\n            A recursive function definition has one or more base cases, meaning input(s) for which the function produces a result trivially (without recurring), and one or more recursive cases, meaning input(s) for which the program recurs (calls itself). For example, the factorial function can be defined recursively by the equations 0! = 1 and, for all n > 0, n! = n(n − 1)!. Neither equation by itself constitutes a complete definition; the first is the base case, and the second is the recursive case. Because the base case breaks the chain of recursion, it is sometimes also called the "terminating case".\\n\r\n            The job of the recursive cases can be seen as breaking down complex inputs into simpler ones. In a properly designed recursive function, with each recursive call, the input problem must be simplified in such a way that eventually the base case must be reached. (Functions that are not intended to terminate under normal circumstances—for example, some system and server processes—are an exception to this.) Neglecting to write a base case, or testing for it incorrectly, can cause an infinite loop.\\n\r\n            For some functions (such as one that computes the series for e = 1/0! + 1/1! + 1/2! + 1/3! + ...) there is not an obvious base case implied by the input data; for these one may add a parameter (such as the number of terms to be added, in our series example) to provide a \'stopping criterion\' that establishes the base case. Such an example is more naturally treated by co-recursion, where successive terms in the output are the partial sums; this can be converted to a recursion by using the indexing parameter to say "compute the nth term (nth partial sum)',
    youtubeLink: 'https://www.youtube.com/watch?v=Mv9NEXX1VHc&feature=emb_title',
    activeUsers: 1,
    approved: true,
    recommendedLessons: [
        {
            id: 5,
            title: 'Loops',
            description: 'Learn how to utilise Loops in your programs.',
            theory:
                'The Java for loop is used to iterate a part of the program several times. If the number of iteration is fixed, it is recommended to use for loop.\r\n\r\nThere are three types of for loops in java.\r\n\r\nSimple For Loop\r\nFor-each or Enhanced For Loop\r\nLabeled For Loop\r\nJava Simple For Loop\r\nA simple for loop is the same as C/C++. We can initialize the variable, check condition and increment/decrement value. It consists of four parts:\r\n\r\nInitialization: It is the initial condition which is executed once when the loop starts. Here, we can initialize the variable, or we can use an already initialized variable. It is an optional condition.\r\nCondition: It is the second condition which is executed each time to test the condition of the loop. It continues execution until the condition is false. It must return boolean value either true or false. It is an optional condition.\r\nStatement: The statement of the loop is executed each time until the second condition is false.\r\nIncrement/Decrement: It increments or decrements the variable value. It is an optional condition.',
            youtubeLink: 'https://www.youtube.com/watch?v=guXgBe2wvEA',
            activeUsers: 1,
            approved: true,
            recommendedLessons: [],
            language: {
                id: 1,
                language: 'java',
                imagePath: 'JavaImg',
                iconPath: 'JavaIcon',
            },
        },
    ],
};
