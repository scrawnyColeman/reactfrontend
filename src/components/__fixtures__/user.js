export const user = {
    id: 1,
    username: 'scrawnycoleman',
    email: 'sample@mail.com',
    passwordHash: '$2y$12$iIvpWDkdxGrRYij/2mo0c.1bnaLO5G.7pqq1eVhEoPRtVKb7nLeBa',
    dateOfBirth: '1997-06-03T00:00:00.000+00:00',
    userRole: {
        id: 1,
        roleName: 'user',
    },
};
