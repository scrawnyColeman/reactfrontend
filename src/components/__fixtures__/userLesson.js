export const userLesson = {
    id: 1,
    lesson: {
        id: 1,
        title: 'Recursion',
        description:
            'Learn recursive functions to ace all of your coding interviews. We recommend that all users complete our loops and conditionals lesson before beginning here.',
        theory:
            'A common computer programming tactic is to divide a problem into sub-problems of the same type as the original, solve those sub-problems, and combine the results. This is often referred to as the divide-and-conquer method; when combined with a lookup table that stores the results of solving sub-problems (to avoid solving them repeatedly and incurring extra computation time), it can be referred to as dynamic programming or memoization.\\n\r\n            A recursive function definition has one or more base cases, meaning input(s) for which the function produces a result trivially (without recurring), and one or more recursive cases, meaning input(s) for which the program recurs (calls itself). For example, the factorial function can be defined recursively by the equations 0! = 1 and, for all n > 0, n! = n(n − 1)!. Neither equation by itself constitutes a complete definition; the first is the base case, and the second is the recursive case. Because the base case breaks the chain of recursion, it is sometimes also called the "terminating case".\\n\r\n            The job of the recursive cases can be seen as breaking down complex inputs into simpler ones. In a properly designed recursive function, with each recursive call, the input problem must be simplified in such a way that eventually the base case must be reached. (Functions that are not intended to terminate under normal circumstances—for example, some system and server processes—are an exception to this.) Neglecting to write a base case, or testing for it incorrectly, can cause an infinite loop.\\n\r\n            For some functions (such as one that computes the series for e = 1/0! + 1/1! + 1/2! + 1/3! + ...) there is not an obvious base case implied by the input data; for these one may add a parameter (such as the number of terms to be added, in our series example) to provide a \'stopping criterion\' that establishes the base case. Such an example is more naturally treated by co-recursion, where successive terms in the output are the partial sums; this can be converted to a recursion by using the indexing parameter to say "compute the nth term (nth partial sum)',
        youtubeLink: 'https://www.youtube.com/watch?v=Mv9NEXX1VHc&feature=emb_title',
        activeUsers: 1,
        approved: true,
        recommendedLessons: [
            {
                id: 5,
                title: 'Loops',
                description: 'Learn how to utilise Loops in your programs.',
                theory:
                    'The Java for loop is used to iterate a part of the program several times. If the number of iteration is fixed, it is recommended to use for loop.\r\n\r\nThere are three types of for loops in java.\r\n\r\nSimple For Loop\r\nFor-each or Enhanced For Loop\r\nLabeled For Loop\r\nJava Simple For Loop\r\nA simple for loop is the same as C/C++. We can initialize the variable, check condition and increment/decrement value. It consists of four parts:\r\n\r\nInitialization: It is the initial condition which is executed once when the loop starts. Here, we can initialize the variable, or we can use an already initialized variable. It is an optional condition.\r\nCondition: It is the second condition which is executed each time to test the condition of the loop. It continues execution until the condition is false. It must return boolean value either true or false. It is an optional condition.\r\nStatement: The statement of the loop is executed each time until the second condition is false.\r\nIncrement/Decrement: It increments or decrements the variable value. It is an optional condition.',
                youtubeLink: 'https://www.youtube.com/watch?v=guXgBe2wvEA',
                activeUsers: 1,
                approved: true,
                recommendedLessons: [],
                language: {
                    id: 1,
                    language: 'java',
                    imagePath: 'JavaImg',
                    iconPath: 'JavaIcon',
                },
            },
            {
                id: 6,
                title: 'Conditionals',
                description: 'Learn to use if statements to create intelligent programs that can make decisions.',
                theory:
                    'Relational operators are used to check conditions like whether two values are equal, or whether one is greater than the other. The following expressions show how they are used:\r\n\r\nx == y          // x is equal to y\r\nx != y          // x is not equal to y\r\nx > y           // x is greater than y\r\nx < y           // x is less than y\r\nx >= y          // x is greater than or equal to y\r\nx <= y          // x is less than or equal to y\r\nThe result of a relational operator is one of two special values, true or false. These values belong to the data type boolean; in fact, they are the only boolean values.\r\n\r\n\r\nYou are probably familiar with these operations, but notice that the Java operators are different from the mathematical symbols like =, ≠, and ≤. A common error is to use a single = instead of a double ==. Remember that = is the assignment operator, and == is a comparison operator. Also, there is no such thing as the =< or => operators.\r\n\r\nThe two sides of a relational operator have to be compatible. For example, the expression 5 < "6" is invalid because 5 is an int and "6" is a String. When comparing values of different numeric types, Java applies the same conversion rules we saw previously with the assignment operator. For example, when evaluating the expression 5 < 6.0, Java automatically converts the 5 to 5.0.\r\n\r\nMost relational operators don’t work with strings. But confusingly, == and != do work with strings – they just don’t do what you expect. We’ll explain what they do later; in the meantime, don’t use them with strings. Instead, you should use the equals method:\r\n\r\nString fruit1 = "Apple";\r\nString fruit2 = "Orange";\r\nSystem.out.println(fruit1.equals(fruit2));\r\nThe result of fruit1.equals(fruit2) is the boolean value false.\r\n\r\nRelational operators are used to check conditions like whether two values are equal, or whether one is greater than the other. The following expressions show how they are used:\r\n\r\nx == y          // x is equal to y\r\nx != y          // x is not equal to y\r\nx > y           // x is greater than y\r\nx < y           // x is less than y\r\nx >= y          // x is greater than or equal to y\r\nx <= y          // x is less than or equal to y\r\nThe result of a relational operator is one of two special values, true or false. These values belong to the data type boolean; in fact, they are the only boolean values.\r\n\r\n\r\nYou are probably familiar with these operations, but notice that the Java operators are different from the mathematical symbols like =, ≠, and ≤. A common error is to use a single = instead of a double ==. Remember that = is the assignment operator, and == is a comparison operator. Also, there is no such thing as the =< or => operators.\r\n\r\nThe two sides of a relational operator have to be compatible. For example, the expression 5 < "6" is invalid because 5 is an int and "6" is a String. When comparing values of different numeric types, Java applies the same conversion rules we saw previously with the assignment operator. For example, when evaluating the expression 5 < 6.0, Java automatically converts the 5 to 5.0.\r\n\r\nMost relational operators don’t work with strings. But confusingly, == and != do work with strings – they just don’t do what you expect. We’ll explain what they do later; in the meantime, don’t use them with strings. Instead, you should use the equals method:\r\n\r\nString fruit1 = "Apple";\r\nString fruit2 = "Orange";\r\nSystem.out.println(fruit1.equals(fruit2));\r\nThe result of fruit1.equals(fruit2) is the boolean value false.\r\n\r\n',
                youtubeLink: 'https://www.youtube.com/watch?v=fsTC0Mqy0EU',
                activeUsers: 1,
                approved: false,
                recommendedLessons: [],
                language: {
                    id: 1,
                    language: 'java',
                    imagePath: 'JavaImg',
                    iconPath: 'JavaIcon',
                },
            },
        ],
        language: {
            id: 2,
            language: 'javascript',
            imagePath: 'JSImg',
            iconPath: 'JSIcon',
        },
    },
    completed: false,
};
