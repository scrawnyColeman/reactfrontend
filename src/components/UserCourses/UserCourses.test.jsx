import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import UserCourses from './component';
import { userLesson } from '../__fixtures__/userLesson';

Enzyme.configure({ adapter: new Adapter() });

describe('UserCourses component', () => {
    const onClick = jest.fn();

    test('component matches snapshot', () => {
        const wrapper = shallow(<UserCourses courses={[userLesson]} onClick={onClick} />);
        expect(wrapper).toMatchSnapshot();
    });
});
