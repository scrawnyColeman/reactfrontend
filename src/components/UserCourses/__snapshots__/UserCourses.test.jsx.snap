// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCourses component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    Pick up where you left off...
  </styled.div>
  <Slider
    dots={true}
    infinite={true}
    responsive={
      Array [
        Object {
          "breakpoint": 1280,
          "settings": Object {
            "dots": true,
            "infinite": true,
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={2}
    speed={500}
  >
    <styled.div
      key="1"
    >
      <styled.div>
        <styled.img
          src="jsimage.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            ReactJS
          </p>
          <p>
            [
            JAVASCRIPT
            ]
          </p>
        </styled.span>
        <styled.p>
          Learn React, an open-source JavaScript library, for building fast webapps and clever User interfaces.
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="2"
    >
      <styled.div>
        <styled.img
          src="javaheader.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            Lambda Functions
          </p>
          <p>
            [
            JAVA
            ]
          </p>
        </styled.span>
        <styled.p>
          Produce concise lambda functions to enhance your functional programming skills. It is recommended that users complete our interfaces lesson first.
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.div>
        <styled.img
          src="python image.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            Recursion
          </p>
          <p>
            [
            PYTHON
            ]
          </p>
        </styled.span>
        <styled.p>
          Learn recursive functions to ace all of your coding interviews. We recommend that all users complete our loops and conditionals lesson before beginnin
          ...
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="4"
    >
      <styled.div>
        <styled.img
          src="javaheader.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            Polymorphism
          </p>
          <p>
            [
            JAVA
            ]
          </p>
        </styled.span>
        <styled.p>
          Meaning to take many forms, polymorphism is among the four pillars of Object-oriented programming; a must know for all programmers.
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="5"
    >
      <styled.div>
        <styled.img
          src="python image.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            Control Flow and Conditionals
          </p>
          <p>
            [
            PYTHON
            ]
          </p>
        </styled.span>
        <styled.p>
          Learn how to write a program that is able to do hundreds of hours of human cognition in seconds through repetition and scripted decision making
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="6"
    >
      <styled.div>
        <styled.img
          src="python image.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            Classes and Objects
          </p>
          <p>
            [
            PYTHON
            ]
          </p>
        </styled.span>
        <styled.p>
          Learn the fundamental elements that make up the world of Object-Oriented Programming.
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="7"
    >
      <styled.div>
        <styled.img
          src="jsimage.jpg"
        />
      </styled.div>
      <styled.div>
        <styled.span>
          <p>
            ES6 Arrow Functions
          </p>
          <p>
            [
            JAVASCRIPT
            ]
          </p>
        </styled.span>
        <styled.p>
          Learn arrow functions - introduced with the ECMAScript 6 - and discover how to write shorter, more concise functions with more capable than before.
        </styled.p>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="Go"
          variant="outlined"
        />
      </styled.div>
    </styled.div>
  </Slider>
</styled.div>
`;
